client/src/pages/UserProfile.jsx

```diff
@@ -63,8 +84,19 @@ export default function UserProfile() {
 
   const userProfileChange = () => {
     setUpdateMode(false);
-    console.log(firstname);
-    console.log(username);
+    setUpdateUser(false);
+    axios.put(`/user/${user._id}/updateUserInfo`, {
+      firstname:firstname,
+      lastname:lastname,
+      description: userDescription, 
+      email: userEmail,
+      profilePic: profilePic,
+    }).then((response) => {
+      if (response) {
+        setUpdateUser(true);
```

Comment:
We should update the user in context here as well or have the context refresh that info so that it consistently receives the same object.